palindrom-client.html

```diff
@@ -23,577 +20,449 @@
+            text-decoration: none!important;
+        }
+
+        @keyframes rotateHG {
+            100% {
+                transform: rotate(360deg)
+            }
+        }
+
+        .tappable {
+            cursor: pointer;
+        }
+
+        .tappable span {
+            text-decoration: underline;
+        }
+    </style>
+    <div class="box-container reconnect-pane" hidden>
+        <div class="box notice tappable" on-tap="reconnectNow">
+            <strong>
+                <span class="hourglass">⏳</span> Not connected.
+            </strong>
+            Connecting in <span class="reconnection-seconds">5</span>s...
+            <span>Retry now</span>
+        </div>
+    </div>
+    <div class="box-container reconnecting-now-pane" hidden>
+        <div class="box notice">
+            <strong>
+                <span class="hourglass">⏳</span> Connecting now...
+            </strong>
+        </div>
+    </div>
+    <div class="box-container error-pane" hidden>
+        <div class="box error tappable">
+            <span class="hourglass">⏳</span> Connection error.</strong> See console for details.
+            <a class="reloading-in-message">
+                Reloading page in <span class="reconnection-seconds">5</span>...
+                • <span on-tap="cancelReloading">Do not reload</span> •
+            </a>
+            <span on-tap="reload">Reload now</span>
+        </div>
+    </div>
+    <div class="box-container generic-error-pane" hidden>
+        <div class="box error tappable" on-tap="reload">
+            <strong>⚠️ An error has occurred.</strong> See console for details.
+            <span>Click here to reload</span>
+        </div>
+    </div>
+</template>
+<script>      
+
+
+    (function() {
+
+        const template = document.currentScript.ownerDocument.querySelector('#palindrom-client');
+
+        const defaultValues = {
+            "useWebSocket": true,
+            "debug": false,
+            "localVersionPath": '/_ver#c$',
+            "remoteVersionPath": '/_ver#s',
+            "ot": true,
+            "purity": false,
+            "pingIntervalS": 60,
+            "path": '/',
+            "devToolsOpen": false,
+            "fatalErrorReloadAfterS": 5
+        }
+
+        class PalindromClient extends HTMLElement {
+            static get observedAttributes() {
+                return ['remote-url', 'use-web-socket', 'debug', 'local-version-path', 'remote-version-path', 'ot', 'purity', 'listen-to', 'ping-interval-s', 'purity', 'fatal-error-reload-after-s', 'path'];
             }
 
-            .box-container[hidden] {
-                display: none;
+            constructor() {
+                super();
+                Object.assign(this, defaultValues);
+                for (const attrib of this.attributes) {
+                    const prop = kebebCaseToCamelCase(attrib.name);
+                    if (PalindromClient.observedAttributes.includes(attrib.name)) {
```

Comment:
Wouldn't it be cleaner to specify setters for  props, and `attributesChangedCallback`

Reply:
Yup. I just forgot tbh.